window.js

```diff
@@ -0,0 +1,371 @@
+    // include class attribute in output
+    self.className.length && !self.getAttribute('class') && props.push({name:'class', value: self.className})
+    return props ? _stringify(props) : '';
+  }
+
+  function _isProperty(key) {
+      var types = ['string','boolean','number']
+      for (var i=0; i<=types.length;i++) {
+        if (self.hasOwnProperty(key) &&
+            types[i] === typeof self[key] &&
+            key !== 'nodeName' &&
+            key !== 'nodeType' &&
+            key !== 'className'
+            ) return true;
+      }
+  }
+
+  var attrs = this.style.cssText ? this.attributes.concat([{name: 'style'}]) : this.attributes;
+
+  a.push('<'+this.nodeName + _propertify() + _stringify(attrs) + _dataify(this.dataset) +'>')
+
+  if (!VOID_ELEMENTS[this.nodeName.toUpperCase()]){
+    a.push(this.innerHTML)
+    a.push('</'+this.nodeName+'>')
+  }
+
+  return a.join('')
+})
+
+Element.prototype.__defineGetter__('textContent', function () {
+  var s = ''
+  this.childNodes.forEach(function (e) {
+    s += e.textContent
+  })
+  return s
+})
+
+Element.prototype.addEventListener = function(t, l) {}
+
+function escapeHTML(s) {
+  return String(s)
+      .replace(/&/g, '&amp;')
+      .replace(/</g, '&lt;')
+      .replace(/>/g, '&gt;');
+}
+
+function escapeAttribute(s) {
+  return escapeHTML(s).replace(/"/g, '&quot;')
+}
+
+Element.prototype.nodeValue = null;
+
+function Text(){}
+
+Text.prototype.nodeType = 3;
+
+Text.prototype.nodeName = '#text';
+
+Text.prototype.__defineGetter__('textContent', function() {
+  return escapeHTML(this.value || '');
+})
+
+Text.prototype.__defineSetter__('textContent', function(v) {
+  this.value = v
+})
+
+Text.prototype.__defineGetter__('nodeValue', function() {
+  return escapeHTML(this.value || '');
+})
+
+Text.prototype.__defineSetter__('nodeValue', function(v) {
+  this.value = v
+})
+
+Text.prototype.replaceData = function(offset, length, str) {
+  this.value = this.value.slice(0, offset) + str + this.value.slice(offset + length);
+};
+
+function Comment(){}
+
+Comment.prototype.nodeType = 8;
+
+Comment.prototype.nodeName = '#comment';
+
+Comment.prototype.__defineGetter__('data', function() {
+  return this.value
+})
+
+Comment.prototype.__defineSetter__('data', function(v) {
+  this.value = v
+})
+
+Comment.prototype.__defineGetter__('outerHTML', function() {
+  return '<!--' + escapeHTML(this.value || '') + '-->'
+})
+
+Comment.prototype.__defineGetter__('nodeValue', function() {
+  return escapeHTML(this.value || '');
+})
+
+Comment.prototype.__defineSetter__('nodeValue', function(v) {
+  this.value = v
+})
+
+function defineParentNode (obj) {
+  obj.__defineGetter__('parentNode', function () { return this.parentElement });
+}
+
+defineParentNode(Element.prototype)
+defineParentNode(Comment.prototype)
+defineParentNode(Text.prototype)
+defineParentNode(Node.prototype)
+
+module.exports = { Document, Node, Element, Comment, Text, document: new Document() };
```

Comment:
NB this object literal shorthand means you'll need node.js v4 or above to use `html-element`. I personally don't see a problem with it but I don't know how @1N50MN14 feels about backwards compatibility with, e.g. v0.12. If you do want to deprecate support for old versions, I'd suggest indicating the requirements in `package.json` with the `engines` field, and a major version bump for `html-element`.